app/_src/guides/consumer-producer-policies.md

```diff
@@ -0,0 +1,262 @@
+  labels:
+    kuma.io/sidecar-injection: enabled" | kubectl apply -f -
+```
+
+Now we can create deployment we will be using to communicate with our demo-app from a first-consumer namespace:
+
+```shell
+kubectl run consumer --image nicolaka/netshoot -n first-consumer --command -- /bin/bash -c "while true; do ping localhost; sleep 60;done"
+```
+
+and from the second-consumer namespace:
+
+```shell
+kubectl run consumer --image nicolaka/netshoot -n second-consumer --command -- /bin/bash -c "while true; do ping localhost; sleep 60;done"
+```
+
+You can make now make a couple of requests to our demo app to check if everything is working: 
+
+```shell
+kubectl exec -n first-consumer consumer -- curl --no-progress-meter demo-app.kuma-demo.mesh:5000/counter
+```
+
+You should see something similar to:
+
+```json
+{"counter":"0","zone":"local","err":null}
+```
+
+## Namespace scoped policies
+
+Now that we have our setup we can start playing with policies. Let's create simple MeshTimeout policy in `kuma-demo` namespace:
+
+```shell
+echo "apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: producer-timeout
+  namespace: kuma-demo
+  labels:
+    kuma.io/mesh: default
+    kuma.io/origin: zone
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: demo-app
+      default:
+        http:
+          requestTimeout: 1s" | kubectl apply -f -
+```
+
+We can now inspect applied policy labels:
+
+```shell
+kubectl get meshtimeout -n kuma-demo producer-timeout -o jsonpath='{.metadata.labels}'
+```
+
+We should see output like:
+
+```json
+{
+  "k8s.kuma.io/namespace": "kuma-demo",
+  "kuma.io/env": "kubernetes",
+  "kuma.io/mesh": "default",
+  "kuma.io/origin": "zone",
+  "kuma.io/policy-role": "producer",
+  "kuma.io/zone": "default"
+}
+```
+
+{{site.mesh_product_name}} is adding custom labels to our policy. The one that interests us most at the moment is `kuma.io/policy-role`.
+This is new label that indicates policy role. Possible values of this label are:
+
+- `system` - policies applied in system namespace
+- `workload-owner` - policies that specify `spec.from` section
+- `consumer` - policies targeting MeshServices from different namespace
+- `producer` - policies targeting MeshServices from the same namespace
+
+### Producer consumer model
+
+With namespace scoped policies we've introduced producer/consumer model for policies. 
+
+**Producer** is someone who authors and operates service. Producer can create default policies that will be used when communicating with service. 
+Producer policies will be created in the same zone as MeshService they target. Producer policies will be synced to other zones. 
+
+**Consumer** is someone that utilizes a service by communicating with it. Consumer policies will be applied in consumer 
+namespace and will target MeshService from different namespace. Consumer policy will take effect only in consumer namespace.
+Consumer policies will override producer policies.
```

Comment:
Does it make sense to explain with a mermaid diagram which sidecars is getting updated in either case?

Reply:
I don't this there is good mermaid diagram type to show this. But I was playing with making a diagram that would make sense and I came up with this: 
![Screenshot 2024-10-01 at 11 55 42](https://github.com/user-attachments/assets/b16f97cc-ce5b-4532-8977-363e1d5cee3a)
Let me know what you think about it

Reply:
Somehow I feel uncomfortable seeing server on the left and client on the right 😄 IDK is it just me or `Client --> Server` has more familiar look to it?

Reply:
You can use mermaid subgraph: https://mermaid.js.org/syntax/flowchart.html#subgraphs

Reply:
Here's an example of a draft I was working on for something else:

```mermaid
flowchart LR

clients@{ shape: processes, label: "clients" }
servers@{ shape: processes, label: "servers"}

subgraph Data-plane
   app
   subgraph data-plane proxy/Envoy
   inbounds
   outbounds
   end
end

inbounds -.local traffic.-> app
app -.local traffic.-> outbounds

clients --> inbounds
outbounds --> servers
```

I think you could show your setup with your app and the 2 different consumers and show how producer/consumer policies affect different sutff (the same base graph would evolve progressively).

Reply:
added diagrams